superset/db_engine_specs/presto.py

```diff
@@ -910,22 +911,26 @@ def _partition_query(cls, table_name, limit=0, order_by=None, filters=None):
     @classmethod
     def where_latest_partition(cls, table_name, schema, database, qry, columns=None):
         try:
-            col_name, value = cls.latest_partition(
+            col_names, values = cls.latest_partition(
                 table_name, schema, database, show_first=True
             )
         except Exception:
             # table is not partitioned
             return False
-        if value is not None:
-            for c in columns:
-                if c.get("name") == col_name:
-                    return qry.where(Column(col_name) == value)
+        if values is not None:
+            is_partitioned = False
+            for i, col_name in enumerate(col_names):
+                if (column.get("name") == col_name for column in columns):
```

Comment:
This is subtly wrong:

```python
>>> columns = [{'name': 'a'}, {'name': 'b'}]
>>> col_name = ['c']
>>> print(bool(column.get("name") == col_name for column in columns))
True
```

You need to wrap the generator expression in `any()`:

```python
>>> print(bool(any(column.get("name") == col_name for column in columns)))
False
```